### PR TITLE
Fix: mgrp child table and documenting transfer and preparation processes

### PR DESCRIPTION
**Pull request by Amna-Alshawa**
## Feature description
- Fixing the way of fetching the children's table in mgrp document.
- Documenting Transfer & Preparation processes

## Areas affected and ensured
All GRD files and `utils.py` and `hook.py` files

## Is there any existing behavior change of other features due to this code change?
No

## Was this feature tested on all the browsers?
  - [x] Chrome
  - [ ] Safari

### CODE REVIEW

File: one_fm/grd/doctype/mgrp/mgrp.py [hunk 1]
```diff
@@ -5,26 +5,32 @@
...
+		self.get_children_table()
 		self.set_grd_values()
 		self.set_status()
-		if not self.end_of_service_attachment:
+		self.set_progress()
+		if self.status == "Cancellation" and not self.end_of_service_attachment:
 			self.set_resignation_attachment()
 		if not self.date_of_application:
-			self.set_date_of_applicantion_value()
+			self.db_set('date_of_application',date.today())
```
Comment (pjamsheer) [inline one_fm/grd/doctype/mgrp/mgrp.py]: On validate you can just put `self.date_of_application = today()`, today can be imported from `frappe.utils`

File: one_fm/hooks.py [hunk 2]
```diff
@@ -218,9 +218,6 @@
 		"before_submit": "one_fm.purchase.utils.before_submit_purchase_receipt",
 		"on_submit": "one_fm.one_fm.doctype.customer_asset.customer_asset.on_purchase_receipt_submit"
 	},
-	"ToDo": {
-		"after_insert": "one_fm.grd.utils.todo_after_insert"
-	},
```
Comment (pjamsheer) [inline one_fm/hooks.py:223]: Need this to be preserved

File: one_fm/grd/utils.py [hunk 3]
```diff
@@ -12,12 +12,6 @@
 from dateutil.relativedelta import relativedelta
 from one_fm.api.notification import create_notification_log
 
-def todo_after_insert(doc, method):
-    doctypes = ['Work Permit', 'Medical Insurance']
-    if doc.reference_type in doctypes and doc.reference_name and doc.owner and 'GRD Operator' in frappe.get_roles(doc.owner):
-        if not frappe.db.get_value(doc.reference_type, doc.reference_name, 'grd_operator'):
-            frappe.db.set_value(doc.reference_type, doc.reference_name, 'grd_operator', doc.owner)
-
```
Comment (pjamsheer) [inline one_fm/grd/utils.py:20]: Needs to be preserved this

File: one_fm/hiring/utils.py [hunk 4]
```diff
@@ -497,38 +503,44 @@ def set_mandatory_feilds_in_employee_for_Kuwaiti(doc,method):
...
+                for row in table:
+                    children = doc.append('children_details',{})
+                    children.child_name = row['child_name']
+                    children.child_name_in_arabic = row['child_name_in_arabic']
+                    children.age = row['age']
+                    children.work_status = row['work_status']
+                    children.married = row['married']
+                    children.health_status = row['health_status']
+                frappe.db.commit()
+
```
Comment (pjamsheer) [inline one_fm/hiring/utils.py]: Why 2 for loops, from first loop itself you can set the values to doc child table ?

File: one_fm/grd/utils.py (hunk 3 above)
Comment (Amna-Alshawa) [reply one_fm/grd/utils.py:20]: @pjamsheer no need to be stored in todo list as there will be a reminder for the records through their emails

File: one_fm/hooks.py (hunk 2 above)
Comment (Amna-Alshawa) [reply one_fm/hooks.py:223]: @pjamsheer no need to be stored in todo list.

**Review by pjamsheer — APPROVED** [review]

**Pull request merged**